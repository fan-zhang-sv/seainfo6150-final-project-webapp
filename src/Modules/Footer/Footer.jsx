import React, { useState } from "react";
import styles from "./Footer.module.css";
import { ccToCountryName } from "../../Modules/countryCodes";

const Footer = () => {
  const [submittedForm, setSubmittedForm] = useState();
  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
    console.log(data.get("selectedCategory"));
  }

  return (
    <div className={styles.Footer}>
      <div className={styles.FormContent}>
        <h2>Subscribe</h2>
        {submittedForm ? (
          <div>
            <h5>
              {submittedForm.get("title")} {submittedForm.get("name")}, you have
              successfully subscribed categoriy{" "}
              {submittedForm.get("selectedCategory")} from{" "}
              {ccToCountryName[submittedForm.get("country")]}.
            </h5>
          </div>
        ) : (
          <form onSubmit={onSubmit}>
            <div>
              <h5>Please tell us your information.</h5>
              <label htmlFor="titleID">Title</label>
              <select name="title" id="titleID">
                <option value="Mr">Mr</option>
                <option value="Miss">Miss</option>
                <option value="Mrs">Mrs</option>
                <option value="Ms">Ms</option>
                <option value="Mx">Mx</option>
              </select>
              <label htmlFor="nameID">Name</label>
              <input type="text" name="name" required id="nameID" />
              <div className={styles.BreakLine} />
              <label htmlFor="emailID">Email</label>
              <input type="email" name="email" required id="emailID" />
            </div>

            <div>
              <h5>Please customize subscription.</h5>
              <label htmlFor="countryID">Country</label>
              <select name="country" id="countryID">
                <option value="us">United States</option>
                <option value="ca">Canada</option>
                <option value="gb">United Kingdom</option>
                <option value="au">Australia</option>
                <option value="cn">China</option>
                <option value="jp">Japan</option>
              </select>
              <div className={styles.BreakLine} />
              <label>
                <input type="radio" name="selectedCategory" value="Local" />
                Local
              </label>
              <label>
                <input type="radio" name="selectedCategory" value="Business" />
                Business
              </label>
              <label>
                <input
                  type="radio"
                  name="selectedCategory"
                  value="Entertainment"
                />
                Entertainment
              </label>
              <label>
                <input
                  type="radio"
                  name="selectedCategory"
                  value="Politics/Society"
                />
                Politics/Society
              </label>
              <label>
                <input type="radio" name="selectedCategory" value="Sports" />
                Sports
              </label>
              <label>
                <input type="radio" name="selectedCategory" value="World" />
                World
              </label>
            </div>
            <input
              className={styles.SubmitButton}
              type="submit"
              value="Submit"
            />
          </form>
        )}
      </div>
      <h6>
        <span>&copy; Fan "Felix" Zhang 2020</span>
      </h6>
    </div>
  );
};

export default Footer;
