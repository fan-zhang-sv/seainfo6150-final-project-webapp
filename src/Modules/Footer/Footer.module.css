.Footer {
  width: 100%;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;
}

.Footer h2 {
  font-family: "DM Serif Display", serif;
  font-size: 1.8rem;
  margin-top: 30px;
  margin-bottom: 10px;
  border-top: 2px solid #000;
  padding-top: 20px;
}

.FormContent {
  width: 100%;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  background-color: #fff;
  padding-bottom: 25px;
  border-bottom: 1px solid #000;
}

.BreakLine {
  display: block;
  margin-bottom: 6px;
}

.SubmitButton {
  margin-top: 15px;
}

h6 {
  font-family: "Fira Sans", sans-serif;
  font-size: 0.6rem;
  color: #000;
  margin-top: 20px;
  font-weight: 400;
}

/* Centered section inside the top navigation */
.FormContent a {
  font-family: "Fira Sans", sans-serif;
  font-size: 1rem;
  color: #000;
  margin: 0.4rem;
}

@media screen and (min-width: 768px) {
  .FormContent {
    width: 80%;
  }
  .BreakLine {
    display: none;
  }
}
