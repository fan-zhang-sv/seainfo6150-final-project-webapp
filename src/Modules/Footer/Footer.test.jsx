import React from "react";
import Footer from "./Footer";
import { render } from "@testing-library/react";

describe("Footer tests", () => {
  it("renders correctly", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
