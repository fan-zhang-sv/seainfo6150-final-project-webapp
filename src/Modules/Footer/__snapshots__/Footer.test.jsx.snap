// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer tests renders correctly 1`] = `
<div>
  <div
    class="Footer"
  >
    <div
      class="FormContent"
    >
      <h2>
        Subscribe
      </h2>
      <form>
        <div>
          <h5>
            Please tell us your information.
          </h5>
          <label
            for="titleID"
          >
            Title
          </label>
          <select
            id="titleID"
            name="title"
          >
            <option
              value="Mr"
            >
              Mr
            </option>
            <option
              value="Miss"
            >
              Miss
            </option>
            <option
              value="Mrs"
            >
              Mrs
            </option>
            <option
              value="Ms"
            >
              Ms
            </option>
            <option
              value="Mx"
            >
              Mx
            </option>
          </select>
          <label
            for="nameID"
          >
            Name
          </label>
          <input
            id="nameID"
            name="name"
            required=""
            type="text"
          />
          <div
            class="BreakLine"
          />
          <label
            for="emailID"
          >
            Email
          </label>
          <input
            id="emailID"
            name="email"
            required=""
            type="email"
          />
        </div>
        <div>
          <h5>
            Please customize subscription.
          </h5>
          <label
            for="countryID"
          >
            Country
          </label>
          <select
            id="countryID"
            name="country"
          >
            <option
              value="us"
            >
              United States
            </option>
            <option
              value="ca"
            >
              Canada
            </option>
            <option
              value="gb"
            >
              United Kingdom
            </option>
            <option
              value="au"
            >
              Australia
            </option>
            <option
              value="cn"
            >
              China
            </option>
            <option
              value="jp"
            >
              Japan
            </option>
          </select>
          <div
            class="BreakLine"
          />
          <label
            for="localID"
          >
            <input
              id="localID"
              name="selectedCategory"
              type="radio"
              value="Local"
            />
            Local
          </label>
          <label
            for="businessID"
          >
            <input
              id="businessID"
              name="selectedCategory"
              type="radio"
              value="Business"
            />
            Business
          </label>
          <label
            for="entertainmentID"
          >
            <input
              id="entertainmentID"
              name="selectedCategory"
              type="radio"
              value="Entertainment"
            />
            Entertainment
          </label>
          <label
            for="politicsID"
          >
            <input
              id="politicsID"
              name="selectedCategory"
              type="radio"
              value="Politics/Society"
            />
            Politics/Society
          </label>
          <label
            for="sportsID"
          >
            <input
              id="sportsID"
              name="selectedCategory"
              type="radio"
              value="Sports"
            />
            Sports
          </label>
          <label
            for="worldID"
          >
            <input
              id="worldID"
              name="selectedCategory"
              type="radio"
              value="World"
            />
            World
          </label>
        </div>
        <input
          class="SubmitButton"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
    <h6>
      <span>
        Â© Fan "Felix" Zhang 2020
      </span>
    </h6>
  </div>
</div>
`;
